docs/guides/latest/volumesnapshot/deployment-backup.md

```diff
@@ -0,0 +1,262 @@
+## Backup Deployment's Volumes
+
+This tutorial will show you how to use Stash Operator to backup Deployment's volumes using [Volume Snapshot](https://kubernetes.io/docs/concepts/storage/volume-snapshots/) API resources.
+
+##### Before You Begin
+
+* At first, you need to familiarize with the GCE CSI-driver, to enable the Volume Snapshotting feature via feature gate and to install Stah Operator in your cluster. If you are not familiar with that please visit [here](docs/guides/latest/volumesnapshot/backup.md) 
+
+To keep everything isolated, we are going to use a separate namespace called `demo` throughout this tutorial.
+
+```console
+$ kubectl create ns demo
+namespace/demo created
+```
+
+Now, we will create a StorageClass for further uses. 
+Sample StorageClass YAML are given below,
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: standard
+parameters:
+  type: pd-standard
+provisioner: pd.csi.storage.gke.io
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+```
+The [volumeBindingMode](https://kubernetes.io/docs/concepts/storage/storage-classes/#volume-binding-mode) field controls when volume binding and dynamic provisioning should occur. By default, the `Immediate` mode indicates that volume binding and dynamic provisioning occurs once the PVC is created and `WaitForFirstConsumer` mode indicates that volume binding and provisioning occurs until a pod using the PVC is created.
+
+Let's create the StorageClass we have shown above.
+
+```console
+$ kubectl apply -f ./docs/examples/volume-snapshot/storageclass.yaml
+storageclass.storage.k8s.io/standard created
+```
+We will also create a `VolumeSnapshotClass` for further uses.
+
+Sample `VolumeSnapshotClass` crd YAML are given below,
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1alpha1
+kind: VolumeSnapshotClass
+metadata:
+  annotations:
+    snapshot.storage.kubernetes.io/is-default-class: "true"
+  name: default-snapshot-class
+snapshotter: pd.csi.storage.gke.io
+```
+
+Here,
+
+* `metadata.annotations` annotations are used to set default [volumeSnapshotClass](https://kubernetes.io/blog/2018/10/09/introducing-volume-snapshot-alpha-for-kubernetes/)
+* `snapshotter` field to point to determined CSI driver, for GCE CSI driver, it will be `pd.csi.storage.gke.io`.
+
+Let's create the `volumeSnapshotClass` crd we have shown above.
+
+```console
+$ kubectl apply -f ./docs/examples/volume-snapshot/default-volumesnapshotclass.yaml
+volumesnapshotclass.snapshot.storage.k8s.io/default-snapshot-class created
+```
+
+
+
+>Note: YAML files used in this tutorial are stored in [/docs/examples/volume-snapshot](/docs/examples/volume-snapshot) directory of [appscode/stash](https://github.com/stashed/stash) repository.
+
```

Comment:
Create a separate section (**Prepare for VolumeSnapshot**) with the later points.